docs/source/basics/defining_cloud_resources.md

```diff
@@ -5,21 +5,36 @@ the task are provisioned automatically and managed for the user until task
 completion.
 
 Tasks can be annotated with the resources they are expected to consume (eg. CPU,
-RAM, GPU) at runtime and these requests will be fullfilled during the scheduling
+RAM, Storage, GPU) at runtime and these requests will be fullfilled during the scheduling
 process.
 
+## CPU
+The number of cpu cores available to the task. If your task uses more than the alotted
+number of cores, the task can continue running but will be throttled.
+
+## Memory
+The amount of memory available to the task. If your task uses more than the alotted
+amount of memory, the task will crash.
+
+## Storage
+The amount of disc space available to the task. If your task uses more than the alotted
+amount of storage, the task will crash.
+
+## GPU
+The number of GPUs available to the task.
```

Comment:
explain that running out of VRAM will usually trigger an exception and that using more compute power than available will usually just mean you run slowly